File: backend/src/schemas/thread.py
"""Pydantic schemas for threads."""
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime


class MessageCreate(BaseModel):
    content: str


class MessageResponse(BaseModel):
    id: str
    thread_id: str
    role: str
    content: str
    model_used: Optional[str] = None
    tokens_used: Optional[int] = None
    created_at: datetime


class ThreadCreate(BaseModel):
    query: str
    focus_mode: str = "all"
    space_id: Optional[str] = None


class ThreadResponse(BaseModel):
    id: str
    title: str
    focus_mode: str
    model_used: str
    messages: List[MessageResponse] = []
    created_at: datetime
    updated_at: datetime

File: frontend/src/types/thread.ts
export interface Thread {
    id: string;
    userId: string;
    spaceId?: string;
    title: string;
    focusMode: string;
    modelUsed: string;
    messages: Message[];
    createdAt: string;
    updatedAt: string;
}

export interface Message {
    id: string;
    threadId: string;
    role: "user" | "assistant";
    content: string;
    sources: Source[];
    modelUsed?: string;
    tokensUsed?: number;
    createdAt: string;
}

export interface Source {
    id: string;
    messageId: string;
    url: string;
    title: string;
    domain: string;
    faviconUrl?: string;
    snippet?: string;
    citationIndex: number;
}
